### ISSUE TITLE
resolve hook doesn't respect tsconfig's compilerOptions

### ISSUE THREAD

**Issue by hardfist** [evanw/esbuild#2265]
It seems that esubuild's buitin resolve will return more information about tsconfig than resolve hook, which will affect transform result such as `UseDefineForClassFields`
https://github.com/evanw/esbuild/blob/master/internal/bundler/bundler.go#L788-L810 resolve hook doesn't support return anything about tsconfig. you can see difference in the demo here https://github.com/hardfist/esbuild-bug/tree/main

we can fix this if we support pass compilerOptions to esbuild directly other than rely on resolve result about tsconfig(which seems rarely useful and inefficient for real world case)

**Comment by evanw** [comment]
Yes, this is currently a limitation of the plugin API. One problem is that esbuild's internals read this information in the resolver for performance, but this does not make for a good plugin API so it has not been exposed. Perhaps your suggestion should be how the existing [`tsconfig` option](https://esbuild.github.io/api/#tsconfig) works.

**Comment by hardfist** [comment]
yes，i think support multi tsconfig is kind of odd and complex，it would be better if we can configure compiler options only by option or root tsconfig
